lib/elixir/test/elixir/kernel/sigils_test.exs

```diff
@@ -121,6 +187,31 @@ bar) in ["foo\\\nbar", "foo\\\r\nbar"]
     assert ~W(Foo.Bar.Baz)a == [:"Foo.Bar.Baz"]
   end
 
+  @compile {:no_warn_undefined, Kernel.SigilsTest.TrailingComma}
+
+  test "sigil w/W warns on trailing comma at compile time, not runtime" do
+    for sigil <- ~w(w W),
+        modifier <- ~w(a s c) do
+      assert capture_err(fn ->
+               Code.compile_string("""
+               defmodule Kernel.SigilsTest.TrailingComma do
+                 def run, do: ~#{sigil}(foo, bar baz)#{modifier}
```

Comment:
We don't need a module for these... we can just call `Code.eval`.

Reply:
I don't understand -- how can I separately test compile time and runtime using `Code.eval`?

Reply:
The warning comes from Macro expansion and both compilation and veal have to expand macros.

Reply:
Yes, but several lines down, I test whether the warning is *not* seen at runtime, and I don't know how I would do that if I used `Code.eval` above.

Reply:
You could call `Macro.expand` and pass it the sigil AST to check it is emitted during expansion. In this case, no eval necessary.